x-pack/plugins/ml/public/data_frame/components/group_by_list/popover_form.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiButton,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+} from '@elastic/eui';
+
+import {
+  dateHistogramIntervalFormatRegex,
+  histogramIntervalFormatRegex,
+  PIVOT_SUPPORTED_GROUP_BY_AGGS,
+} from '../../common';
+
+export type supportedIntervalTypes =
+  | PIVOT_SUPPORTED_GROUP_BY_AGGS.HISTOGRAM
+  | PIVOT_SUPPORTED_GROUP_BY_AGGS.DATE_HISTOGRAM;
+
+export function isIntervalValid(interval: string, intervalType: supportedIntervalTypes) {
+  let valid = true;
+
+  valid = interval !== '';
+
+  if (valid && intervalType === PIVOT_SUPPORTED_GROUP_BY_AGGS.HISTOGRAM) {
```

Comment:
nit/suggestion, this check could be done without the `valid` variable.
```
if(interval !== '') {
  if(intervalType === PIVOT_SUPPORTED_GROUP_BY_AGGS.HISTOGRAM){
    return histogramIntervalFormatRegex.test(interval);
  } else if...
}
return false;
```

Reply:
Good idea, changed in 2d9b55f.